src/versions.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import * as hc from '@actions/http-client'
+
+// https://storage.googleapis.com/dart-archive/channels/stable/release/latest/VERSION
+// {
+//   "date": "2023-02-07",
+//   "version": "2.19.2",
+//   "revision": "e46b4f59490230778e907bde2eedb06b062d31be"
+// }
+
+// Query google storage for the most recent published SDK version for the given
+// channel and flavor.
+export async function latestPublishedVersion(
+  channel: string,
+  flavor: string
+): Promise<string | null> {
```

Comment:
The `null` isn't handled at the call sites, shouldn't we throw instead?

Right now, the action will print "Installing the linux-x64 Dart SDK version null..." and then fail somewhere later (probably when it tries to download from dart-archive, which hopefully doesn't have a file at the URL containing null).